src/test/scala/mesosphere/marathon/storage/migration/MigrationTo18Test.scala

```diff
@@ -0,0 +1,115 @@
+package mesosphere.marathon
+package storage.migration
+
+
+import akka.Done
+import akka.stream.scaladsl.Source
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.storage.store.impl.zk.ZkPersistenceStore
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.task.state.NetworkInfo
+import mesosphere.marathon.state.{PathId, Timestamp}
+import mesosphere.marathon.storage.repository.InstanceRepository
+import org.apache.mesos
+import org.apache.mesos.Protos.NetworkInfo.Protocol
+import org.scalatest.Inspectors
+import play.api.libs.json.{JsObject, JsString, JsValue, Json}
+
+import scala.concurrent.Future
+
+class MigrationTo18Test extends AkkaUnitTest with StrictLogging with Inspectors {
+
+  "Migration to 18" should {
+    "Update instance condition from Created to Provisioned" in {
+
+      Given("two instances in Created state")
+      val f = new Fixture()
+
+      val instanceId1 = Instance.Id.forRunSpec(PathId("/app"))
+      val instanceId2 = Instance.Id.forRunSpec(PathId("/app2"))
+
+      val instance1Json = f.created(instanceId1)
+      val instance1 = instance1Json.as[Instance]
+      val migratedInstance1 = f.setProvisionedCondition(instance1)
+
+      val instance2Json = f.created(instanceId2)
+      val instance2 = instance2Json.as[Instance]
+      val migratedInstance2 = f.setProvisionedCondition(instance2)
+
+      f.instanceRepository.ids() returns Source(List(instanceId1, instanceId2))
+      f.persistenceStore.get[Instance.Id, JsValue](equalTo(instanceId1))(any, any) returns Future(Some(instance1Json))
+      f.persistenceStore.get[Instance.Id, JsValue](equalTo(instanceId2))(any, any) returns Future(Some(instance2Json))
+      f.instanceRepository.store(equalTo(migratedInstance1)) returns Future.successful(Done)
+      f.instanceRepository.store(equalTo(migratedInstance2)) returns Future.successful(Done)
+
+      When("they are migrated")
+      MigrationTo18.migrateInstanceConditions(f.instanceRepository, f.persistenceStore).futureValue
+
+      Then("all updated instances are saved")
+      verify(f.instanceRepository, times(2)).store(any)
```

Comment:
wouldn't it make more sense to verify that store was called for those two concrete instances?

Reply:
Well we have this checked already in 
```
f.instanceRepository.store(equalTo(migratedInstance1)) returns Future.successful(Done)
f.instanceRepository.store(equalTo(migratedInstance2)) returns Future.successful(Done)
```

Reply:
well that's not really the same, right? if someone called store with a different instance two times, the test will still pass? Just that the store will return null instead of future... so maybe some exception would be thrown somewhere .... ?

Reply:
`returns Future.successful(Done)` just means it will return the value if it's called. You still have to verify that it was actually called with `verify(...)`. Additionally, I'd like to add a third instance in non-Created condition and make sure it survives the migration without a change.

Reply:
> if someone called store with a different instance two times, the test will still pass?

no it will throw a NPE in this case.

> returns Future.successful(Done) just means it will return the value if it's called. 
> You still have to verify that it was actually called with verify(...).

If the input arguments are not the same as in the mock, it will throw NPE. I don't understand why do we need to have an additional verification in `verify` since the content is already verified in the mock. Please correct me if I'm wrong.